### DISCUSSION TITLE
Precise behavior of --justdb

### DISCUSSION THREAD

**Discussion by QuantumTux** [rpm-software-management/rpm#3224] [Q&A] [answered]
**RPM v4.16.1** (in RHEL v9)

I have some RPMs that use Triggers and deploy some files

For various reasons, I want to "update" the RPMs - that is, I want to "remove" the "old" ones and "install" the "new" ones. In a few cases, the RPMs are changing names (even though the "old" RPM does exactly the same thing as the "new" RPM).

**There's no real change to the files deployed** - this is actually about fixing some typos/documentation/description mistakes, plus an occasional dependency fix-up. However, in a number of cases, removing (that is, just plain **rpm --erase** or **rpm --upgrade**) the "old" RPM would cause a service interruption to some software on the hosts. So I just want to swap out the RPMs in the DB.

Of **critical** importance is that none of the files deployed by an RPM should be changed, nor should any **%pre**, **%post**, **%preun**, **%postun** or **%trigger** script get executed.

The wording of the **man**  page says

`  --justdb`
`              Update only the database, not the filesystem.`

While it might be reasonable to **_ass/u/me_** that it implies **--noscripts** and **--notriggers**, I prefer to make certain.

So, that is my question: what is the _precise_ behavior of **--just-db** in this context?

**Comment by pmatilai** [comment, marked as answer]
--justdb indeed disables all scriptlets, as you would assume.
